core/templates/components/state-directives/solution-editor/solution-editor.component.spec.ts

```diff
@@ -16,65 +16,135 @@
  * @fileoverview Unit test for Solution Editor Component.
  */
 
-import { importAllAngularServices } from 'tests/unit-test-utils.ajs';
-
-describe('SolutionEditorComponent', () => {
-  let ctrl = null;
-  let $rootScope = null;
-  let $scope = null;
-
-  let EditabilityService = null;
-  let ExplorationHtmlFormatterService = null;
-  let StateSolutionService = null;
-  let StateInteractionIdService = null;
-  let StateCustomizationArgsService = null;
-
-  beforeEach(angular.mock.module('oppia'));
-  importAllAngularServices();
-
-  beforeEach(angular.mock.inject(function($injector, $componentController) {
-    $rootScope = $injector.get('$rootScope');
-    $scope = $rootScope.$new();
-
-    EditabilityService = $injector.get('EditabilityService');
-    StateSolutionService = $injector.get('StateSolutionService');
-    StateInteractionIdService = $injector.get('StateInteractionIdService');
-    EditabilityService = $injector.get('EditabilityService');
-    ExplorationHtmlFormatterService = $injector.get(
-      'ExplorationHtmlFormatterService');
-    StateCustomizationArgsService = $injector.get(
-      'StateCustomizationArgsService');
-
-    ctrl = $componentController('solutionEditor', {
-      $scope: $scope
-    });
-
-    StateSolutionService.savedMemento = {
-      correctAnswer: 'Ans'
-    };
-    StateInteractionIdService.savedMemento = 'TextInput';
-    StateCustomizationArgsService.savedMemento = 'Args';
+import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+import { SolutionEditor } from './solution-editor.component';
+import { EditabilityService } from 'services/editability.service';
+import { SolutionObjectFactory } from 'domain/exploration/SolutionObjectFactory';
+import { StateCustomizationArgsService } from 'components/state-editor/state-editor-properties-services/state-customization-args.service';
+import { ExplorationHtmlFormatterService } from 'services/exploration-html-formatter.service';
+import { StateInteractionIdService } from 'components/state-editor/state-editor-properties-services/state-interaction-id.service';
+import { StateSolutionService } from 'components/state-editor/state-editor-properties-services/state-solution.service';
+
+class mockStateCustomizationArgsService {
+  savedMemento = 'data3';
+}
+class mockStateInteractionIdService {
+  savedMemento = 'data2';
+}
+class mockStateSolutionService {
+  savedMemento = {
+    correctAnswer: 'data1'
+  };
+}
+class mockExplorationHtmlFormatterService {
+  getAnswerHtml(x, y, z): string {
+    return x + y + z;
+  }
+}
+class mockEditabilityService {
```

Comment:
Name for class should start with a capital letter.

Reply:
done